sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobApiTests.java

```diff
@@ -1728,6 +1728,35 @@ public void encryptionDataCaseInsensitivityDownloadToFile(String newKey, Encrypt
         compareFiles(file, outFile, 0, file.length());
     }
 
+    @ParameterizedTest
+    @MethodSource("uploadAndDownloadDifferentRegionLengthSupplier")
+    public void uploadAndDownloadDifferentRegionLength(int regionLength, int dataSize) {
+        ByteBuffer data = getRandomData(dataSize);
+        beac = mockAesKey(getEncryptedClientBuilder(fakeKey, null, ENV.getPrimaryAccount().getCredential(),
+            cc.getBlobContainerUrl(), EncryptionVersion.V2)
+            .blobName(generateBlobName())
+            .blobEncryptionOptions(new BlobEncryptionOptions().setAuthenticatedRegionDataLength(regionLength))
+            .buildEncryptedBlobAsyncClient());
+        beac.uploadWithResponse(new BlobParallelUploadOptions(Flux.just(data.duplicate()))).block();
+        ByteArrayOutputStream plaintextOut = new ByteArrayOutputStream();
+
+        new EncryptedBlobClient(beac).downloadStream(plaintextOut);
```

Comment:
It would also be good to see if we have test cases for the `downloadToFile` method since it has some extra logic on the `downloadStream` that exercise the ranged header expansion that the `downloadStream` method won't use.